machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -0,0 +1,151 @@
+// +openshift:compatibility-gen:level=4
+type MachineConfigNodeList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MachineConfigNode `json:"items"`
+}
+
+// MCOObjectReference holds information about an object the MCO either owns
+// or modifies in some way
+type MCOObjectReference struct {
+	// name describes what this object is called
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+// MachineConfigNodeSpec describes the type of State we are managing
+type MachineConfigNodeSpec struct {
+	// node contains references to the node for this MachineConfigNode
+	// +kubebuilder:validation:Required
+	Node MCOObjectReference `json:"node"`
+
+	// pool contains a reference to the MachineConfigPool that this MachineConfigNode's referenced Node belongs to.
+	// +kubebuilder:validation:Required
+	Pool MCOObjectReference `json:"pool"`
+
+	// configVersion holds the desired config version for the node targeted by this MachineConfigNode resource.
+	// The desired version represents the machine config the node will attempt to update to when it next has the chance.
+	// +kubebuilder:validation:Required
+	ConfigVersion MCNSpecMachineConfigVersion `json:"configVersion"`
+}
+
+// MachineConfigNodeStatus holds the reported information on a particular MachineConfigNode
+type MachineConfigNodeStatus struct {
+	// conditions represent the observations of a MachineConfigNode's current state.
+	// Known .status.conditions.type are: "UpdatePreparing", "UpdateInProgress", "UpdatePostAction", "UpdateCompleting", "Resuming", "Updated",
+	// "Ready", "ComparingMCs", "DrainingNode", "ApplyingFilesAndOS", "CordoningNode", "RebootingNode", and "ReloadingCRIO"
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// observedGeneration represents the generation observed by the controller.
+	// +kubebuilder:validation:Required
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// configVersion describes the current and desired MachineConfig for this node.
+	// The current version represents the current machine config for the node and is updated after a successful update.
+	// The desired version represents the machine config the node will attempt to update to when it next has the chance.
+	// +kubebuilder:validation:Required
+	ConfigVersion MCNStatusMachineConfigVersion `json:"configVersion"`
+}
+
+// MCNStatusMachineConfigVersion holds the current and desired config versions as last updated in the MCN status.
+// When Current != Desired; the MachineConfigOperator is in an upgrade phase and the MachineConfigNode will
+// take account of upgrade related events. Otherwise they will be ignored given that certain operations
+// happen both during the MCO's upgrade mode and the daily operations mode.
```

Comment:
Bear in mind that all of this documentation is customer facing, we try to keep it to the same sort of tone as product docs. I would suggest using prose rather than `!=` to explain the inequality here
```suggestion
// When the current and desired versions are not matched, the machine config pool is processing an upgrade and the machine config node will
// monitor the upgrade process. 
// When the current and desired versions are not match, the machine config node will ignore these events given that certain operations happen both during the MCO's upgrade mode and the daily operations mode.
```
I'm struggling with the last sentence here, why do we ignore/what do we ignore on a day to day basis?

Reply:
what I was trying to say here is that some of these events (specifically node draining) can happen in contexts outside of the MCO. So watching the node all of the time 1) is unnecessary load on the MCO and 2) will result in probable accidental "upgrade events" that are really other people acting on the node. We only want to update our MCN when we are sure the MCO is doing something upgrade related.

Reply:
I changed the comment to your version though